docs/storage/commit_log/commit_log_based_storage_design.md

```diff
@@ -0,0 +1,354 @@
+## Commit-Log based Trillian storage
+
+*Status: Draft*
+
+*Authors: al@google.com, drysdale@google.com, filippo@cloudflare.com*
+
+*Last Updated: 2017-05-11*
+
+
+[TOC]
+
+
+## Objective
+
+A design for an alternative Trillian storage layer which uses a distributed and
+immutable *commit log* as the source of truth for a Trillian Log's contents and
+sequence information, and one or more independent *"readonly"* databases built
+from the commit log to serve queries.
+
+This design allows for:
+
+*   flexibility in scaling Trillian deployments,
+*   easier recovery from corrupt/failed database deployments since operators
+can simply delete it and allow it to rebuild itself from the commit log.
```

Comment:
Deleting is easy, but you'd want to be pretty damn sure that recreating a DB for whatever size a 'typical' log might be can be achieved in < MMD period for the log.
I guess that your frontends could continue to accept submissions and have the sequencer push them to the Kafka topic, but you'd not be able to serve entries (or STHs? what about those) while the local DB replica is being replayed from the commit log.